src/main/java/core/basesyntax/service/ReaderServiceImpl.java

```diff
@@ -0,0 +1,66 @@
+package core.basesyntax.service;
+
+import core.basesyntax.dao.FruitStorageDao;
+import core.basesyntax.model.Fruit;
+import core.basesyntax.model.Operation;
+import java.io.IOException;
+import java.math.BigDecimal;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+
+public class ReaderServiceImpl implements ReaderService {
+    private FruitStorageDao fruitStorageDao;
+    private OperationStrategy operationStrategy;
+
+    public ReaderServiceImpl(FruitStorageDao fruitStorageDao, OperationStrategy operationStrategy) {
+        this.fruitStorageDao = fruitStorageDao;
+        this.operationStrategy = operationStrategy;
+    }
+
+    @Override
+    public void readFromFile(String fileName) {
```

Comment:
```suggestion
    public List<String> readFromFile(String fileName) {
```
щоб було дотримано SRP з СОЛІДа краще щоб цей метод лише за зчитування відповідав а валідацію і мапінг відокремити